nomad/job_endpoint_hooks.go

```diff
@@ -402,7 +409,19 @@ func (v *jobValidate) Validate(job *structs.Job) (warnings []error, err error) {
 	return warnings, validationErrors.ErrorOrNil()
 }
 
-func (v *jobValidate) validateServiceIdentity(s *structs.Service, parent string) error {
+func (v *jobValidate) isEligibleForMultiIdentity() bool {
+	if v.srv == nil || v.srv.serf == nil {
+		return true // handle tests w/o real servers safely
+	}
+	return ServersMeetMinimumVersion(
+		v.srv.Members(), v.srv.Region(), minVersionMultiIdentities, true)
```

Comment:
Would be nice if ServersMeetMinimumVersion was a struct that wrapped up `Members()` and `Region()` and only provided a simple `Meets(version, checkFailed)` interface for us to mock/stub out in tests.

nbd though, nothing to block this work on.

Reply:
I've opened https://github.com/hashicorp/nomad/issues/18931 to follow-up on that when I have a few minutes.